core/trino-main/src/main/java/io/trino/server/security/oauth2/SecretKeyConfig.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.server.security.oauth2;
+
+import io.airlift.configuration.Config;
+import io.airlift.configuration.ConfigDescription;
+import io.airlift.configuration.ConfigSecuritySensitive;
+import io.jsonwebtoken.io.Decoders;
+import io.jsonwebtoken.security.Keys;
+
+import javax.crypto.SecretKey;
+
+import static com.google.common.base.Strings.isNullOrEmpty;
+
+public class SecretKeyConfig
+{
+    private SecretKey secretKey;
```

Comment:
We could default this to a random value so that it's configuration would be optional. Really the only reason you might want to set this value is if you wanted tokens to persist across restarts. I can go either way on this one, but I think it might be convenient.

Reply:
`JweSecretKeysProvider` implements this. I'm not sure if implementing it as a part of config object.